### ISSUE TITLE
LocalProtocolError: Too much data for declared Content-Length

### ISSUE THREAD

**Issue by trondhindenes** [piccolo-orm/piccolo_admin#126]
First of all: Thanks for creating piccolo adn piccolo-admin - it looks really good!!

When using piccolo_admin with Fastapi running in uvicorn, I'm getting this error whenever I delete an object in piccolo-admin.

```
Traceback (most recent call last):
  File "/projects/customer-api/.venv/lib/python3.9/site-packages/uvicorn/protocols/http/h11_impl.py", line 373, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/projects/customer-api/.venv/lib/python3.9/site-packages/uvicorn/middleware/proxy_headers.py", line 75, in __call__
    return await self.app(scope, receive, send)
  File "/projects/customer-api/.venv/lib/python3.9/site-packages/fastapi/applications.py", line 208, in __call__
    await super().__call__(scope, receive, send)
  File "/projects/customer-api/.venv/lib/python3.9/site-packages/starlette/applications.py", line 112, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/projects/customer-api/.venv/lib/python3.9/site-packages/starlette/middleware/errors.py", line 181, in __call__
    raise exc
  File "/projects/customer-api/.venv/lib/python3.9/site-packages/starlette/middleware/errors.py", line 159, in __call__
    await self.app(scope, receive, _send)
  File "/projects/customer-api/.venv/lib/python3.9/site-packages/starlette/exceptions.py", line 82, in __call__
    raise exc
  File "/projects/customer-api/.venv/lib/python3.9/site-packages/starlette/exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "/projects/customer-api/.venv/lib/python3.9/site-packages/starlette/routing.py", line 656, in __call__
    await route.handle(scope, receive, send)
  File "/projects/customer-api/.venv/lib/python3.9/site-packages/starlette/routing.py", line 408, in handle
    await self.app(scope, receive, send)
  File "/projects/customer-api/.venv/lib/python3.9/site-packages/starlette/exceptions.py", line 82, in __call__
    raise exc
  File "/projects/customer-api/.venv/lib/python3.9/site-packages/starlette/exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "/projects/customer-api/.venv/lib/python3.9/site-packages/starlette/middleware/base.py", line 56, in __call__
    await response(scope, receive, send)
  File "/projects/customer-api/.venv/lib/python3.9/site-packages/starlette/responses.py", line 227, in __call__
    await wrap(partial(self.listen_for_disconnect, receive))
  File "/projects/customer-api/.venv/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 567, in __aexit__
    raise exceptions[0]
  File "/projects/customer-api/.venv/lib/python3.9/site-packages/starlette/responses.py", line 223, in wrap
    await func()
  File "/projects/customer-api/.venv/lib/python3.9/site-packages/starlette/responses.py", line 215, in stream_response
    await send({"type": "http.response.body", "body": chunk, "more_body": True})
  File "/projects/customer-api/.venv/lib/python3.9/site-packages/starlette/exceptions.py", line 68, in sender
    await send(message)
  File "/projects/customer-api/.venv/lib/python3.9/site-packages/starlette/exceptions.py", line 68, in sender
    await send(message)
  File "/projects/customer-api/.venv/lib/python3.9/site-packages/starlette/middleware/errors.py", line 156, in _send
    await send(message)
  File "/projects/customer-api/.venv/lib/python3.9/site-packages/uvicorn/protocols/http/h11_impl.py", line 469, in send
    output = self.conn.send(event)
  File "/projects/customer-api/.venv/lib/python3.9/site-packages/h11/_connection.py", line 468, in send
    data_list = self.send_with_data_passthrough(event)
  File "/projects/customer-api/.venv/lib/python3.9/site-packages/h11/_connection.py", line 501, in send_with_data_passthrough
    writer(event, data_list.append)
  File "/projects/customer-api/.venv/lib/python3.9/site-packages/h11/_writers.py", line 58, in __call__
    self.send_data(event.data, write)
  File "/projects/customer-api/.venv/lib/python3.9/site-packages/h11/_writers.py", line 78, in send_data
    raise LocalProtocolError("Too much data for declared Content-Length")
h11._util.LocalProtocolError: Too much data for declared Content-Length
```

Relevant packages:
```
piccolo-admin==0.18.1
  - aiofiles [required: >=0.5.0, installed: 0.7.0]
  - fastapi [required: >=0.62.0, installed: 0.70.0]
    - pydantic [required: >=1.6.2,<2.0.0,!=1.8.1,!=1.8,!=1.7.3,!=1.7.2,!=1.7.1,!=1.7, installed: 1.8.2]
      - typing-extensions [required: >=3.7.4.3, installed: 3.10.0.2]
    - starlette [required: ==0.16.0, installed: 0.16.0]
      - anyio [required: >=3.0.0,<4, installed: 3.3.4]
        - idna [required: >=2.8, installed: 3.3]
        - sniffio [required: >=1.1, installed: 1.2.0]
  - Hypercorn [required: Any, installed: 0.12.0]
    - h11 [required: Any, installed: 0.12.0]
```

**Comment by dantownsend** [comment]
Hi @trondhindenes.

It's a strange one. I can't replicate it locally, but I have an idea what the problem might be.

I found this [similar issue](https://github.com/tiangolo/fastapi/issues/2253) - it looks like the error occurs when an endpoint returns a 204 status code, and the body contains data (which isn't allowed):

https://github.com/piccolo-orm/piccolo_api/blob/dc139c710e23ae8a01bb47c9b267a00245596969/piccolo_api/crud/endpoints.py#L800

I'll remove the body from the response, and will see if that helps.

**Comment by trondhindenes** [comment]
awesome - I patched my local installation with your PR change, and the exception went away.

**Comment by dantownsend** [comment]
The fix is on PyPI now - `piccolo-api==0.28.1`

**Comment by trondhindenes** [comment]
thanks for the quick turnaround!
